railties/lib/rails/commands/server/server_command.rb

```diff
@@ -10,6 +10,33 @@
 
 module Rails
   class Server < ::Rack::Server
+    module HandlerSuggestion
+      include Rails::Command::Behavior
+
+      # Hard-coding a bunch of handlers here as we don't have a public way of
+      # querying them from the Rack::Handler registry.
+      HANDLERS = %w(cgi fastcgi webrick lsws scgi thin puma unicorn)
+
+      def self.message(handler)
+        if handler.in?(HANDLERS)
+          <<~MSG
+            Could not load server "#{handler}". Maybe you need the add it to the Gemfile?
+
+              gem "#{handler}"
+
+            Run `rails server --help` for more options.
+          MSG
+        else
+          suggestions = HANDLERS.sort_by { |h| levenshtein_distance(handler, h) }.map(&:inspect)
```

Comment:
I'm curious. Is the user using the `server` option so much that mistyped check is needed? 
I understand that mistakenly specify server as env. But will not it solve the necessity of option to specify server?

Reply:
@y-yagi doubt it's used often. This tries to give you more context on what actually happens when you pass a positional argument. I'm setting the suggestions here by suggestion (rim-shot). 😅 Anyway, I think it's quite easy to setup the suggestions and we can follow the example in more command options that accept a fixed set of values.